emcc.py

```diff
@@ -2214,7 +2214,8 @@ def compile_source_file(i, input_file):
          not shared.Settings.BOOTSTRAPPING_STRUCT_INFO and \
          not shared.Settings.SIDE_MODULE: # shared libraries/side modules link no C libraries, need them in parent
         extra_files_to_link = system_libs.get_ports(shared.Settings)
-        extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout_=None, stderr_=None, forced=forced_stdlibs)
+        if '-nostdlib' not in newargs and '-nodefaultlibs' not in newargs:
```

Comment:
should this affect ports as well on the previous line? if so it can be part of that `if`

Reply:
I don't think it should effect ports.    I think it should effect system libraries only.